### PR TITLE
[Merged by Bors] - feat: +1 proof that Sorgenfrey plane is not normal

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Separation/NotNormal.lean [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+variable {X : Type u} [TopologicalSpace X] [SeparableSpace X]
+
+/-- Let `s` be a closed set in a separable normal space. If the induced topology on `s` is discrete,
+then `s` has cardinality less than continuum.
+
+The proof follows
+https://en.wikipedia.org/wiki/Moore_plane#Proof_that_the_Moore_plane_is_not_normal -/
+theorem IsClosed.mk_lt_continuum [NormalSpace X] {s : Set X} (hs : IsClosed s)
+    [DiscreteTopology s] : #s < 𝔠 := by
+  -- Proof by contradiction: assum `𝔠 ≤ #s`
```
Comment (jcommelin) [inline Mathlib/Topology/Separation/NotNormal.lean:28]:
```suggestion
  -- Proof by contradiction: assume `𝔠 ≤ #s`
```

File: Mathlib/Topology/Separation/NotNormal.lean [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  haveI := htc.to_subtype
+  -- To obtain a contradiction, we will prove `2 ^ 𝔠 ≤ 𝔠`.
+  refine (Cardinal.cantor 𝔠).not_le ?_
+  calc
+    -- Any function `s → ℝ` is continuous, hence `2 ^ 𝔠 ≤ #C(s, ℝ)`
+    2 ^ 𝔠 ≤ #C(s, ℝ) := by
+      rw [(ContinuousMap.equivFnOfDiscrete _ _).cardinal_eq, mk_arrow, mk_real, lift_continuum,
+        lift_uzero]
+      exact (power_le_power_left two_ne_zero h).trans (power_le_power_right (nat_lt_continuum 2).le)
+    -- By Tietze Extension Theorem, any function `f : C(s, ℝ)` can be extended to `C(X, ℝ)`, hence
```
Comment (jcommelin) [inline Mathlib/Topology/Separation/NotNormal.lean:41]:
```suggestion
    -- By the Tietze Extension Theorem, any function `f : C(s, ℝ)` can be extended to `C(X, ℝ)`, hence
```

**Comment by jcommelin** [comment]
bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**